example/remote/remote.cpp

```diff
@@ -0,0 +1,149 @@
+            lif.tau_m =   2.0;
+            lif.V_th  = -10.0;
+            lif.C_m   =  20.0;
+            lif.E_L   = -23.0;
+            lif.V_m   = -23.0;
+            lif.E_R   = -23.0;
+            lif.t_ref =   0.2;
+            return lif;
+        }
+        std::vector<arb::ext_cell_connection> external_connections_on(arb::cell_gid_type) const override {
+            std::vector<arb::ext_cell_connection> res;
+            // Invent some fictious cells outside of Arbor's realm
+            for (arb::cell_gid_type gid = 0; gid < 10; gid++) {
+                res.emplace_back(arb::cell_remote_label_type{gid}, arb::cell_local_label_type{"tgt"}, weight, delay);
+            }
+            return res;
+        }
+        arb::cell_kind get_cell_kind(arb::cell_gid_type) const override { return arb::cell_kind::lif; }
+        std::vector<arb::probe_info> get_probes(arb::cell_gid_type) const override { return {arb::lif_probe_voltage{}}; }
+};
+
+struct mpi_handle {
+    MPI_Comm global = MPI_COMM_NULL;      // Global communicator; all processes live here
+    int rank = -1, size = -1;             // Global coordinates
+    MPI_Comm local = MPI_COMM_NULL;       // Local communicator; processes of the group live here
+    int local_rank = -1, local_size = -1; // Local coordinates
+    int group = -1;                       // Group index, in {0, 1}
+    MPI_Comm inter = MPI_COMM_NULL;       // Intercommunicator between the groups
+};
+
+mpi_handle setup_mpi();
+
+void sampler(arb::probe_metadata, std::size_t, const arb::sample_record*);
+
+std::vector<std::pair<double, double>> trace;
+
+int main() {
+    // Setup
+    auto T = 1.0;
+    auto dt = 0.025;
+    auto mpi = setup_mpi();
+    if (mpi.group == 1) {
+        auto ctx = arb::make_context({}, mpi.local, mpi.inter);
+        auto rec = remote_recipe();
+        auto sim = arb::simulation(rec, ctx);
+        double mid = sim.min_delay();
+        std::cerr << "[ARB]" << mpi.rank << " " << mpi.local_rank << '\n';
+        std::cerr << "[ARB] Got min delay=" << mid << '\n';
+        sim.add_sampler(arb::all_probes,
+                        arb::regular_schedule(0.05),
+                        sampler);
+        sim.run(T, dt);
+        std::cout << std::fixed << std::setprecision(4);
+        std::cerr << "[ARB] Trace\n";
+        for (const auto& [t, v]: trace) std::cout << " " << t << " " << v << '\n';
+    }
+    else if (mpi.group == 0) {
+        // Simulate external spikes
+        // * We send from a fictious cell with gid=local rank
+        // * The time is a function of rank and epoch.
+        // * We never stop until the other group kills the process ;)
+        std::cerr << "[EXT]" << mpi.rank << " " << mpi.local_rank << '\n';
+        for (int ep = 0;; ++ep) {
+            std::cerr << "[EXT] Epoch " << ep << '\n';
+            arb::cell_gid_type src = mpi.local_rank;
+            auto time = ep*2.0*dt + mpi.local_rank*dt + 0.05;
```

Comment:
Can we calculate the time based on the received message to demonstrate the value of the payload?

Reply:
This is computing fake spikes to send for the next epoch, so yes, we could use the payload, kind of, awkwardly.
For now I just print the bounds.